### PR TITLE
Remove empty string array allocations in DeclarationTreeBuilder

### PR DESCRIPTION
**Pull request by pharring**
I spotted some empty string arrays in a CodeLens memory dump that can be eliminated.

### CODE REVIEW

**Comment by pharring** [comment]
@dotnet-bot retest this please

**Comment by pharring** [comment]
@VSadov @AlekseyTs for review.
Note: This is master only. It's not significant enough to take to Shiproom.

**Comment by VSadov** [comment]
LGTM

File: src/Compilers/VisualBasic/Portable/Declarations/DeclarationTreeBuilder.vb [hunk 1]
```diff
@@ -676,9 +676,17 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
...
 
-            Dim result = results.ToArray()
-            results.Free()
+            ' PERF: The member names collection tends to be long-lived. Use a string array since
+            ' that uses less memory than a HashSet.
+            Dim result As String()
+            If results.Count = 0 Then
+                result = SpecializedCollections.EmptyArray(Of String)
+            Else
+                ReDim result(results.Count - 1)
```
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Portable/Declarations/DeclarationTreeBuilder.vb]: Any particular reason to use ReDim here?

**Comment by AlekseyTs** [comment]
LGTM

File: src/Compilers/VisualBasic/Portable/Declarations/DeclarationTreeBuilder.vb (hunk 1 above)
Comment (pharring) [inline src/Compilers/VisualBasic/Portable/Declarations/DeclarationTreeBuilder.vb]: I thought that's how you create an array in VB. What's the alternative?

Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Portable/Declarations/DeclarationTreeBuilder.vb]: New String (results.Count - 1) {}

Comment (pharring) [inline src/Compilers/VisualBasic/Portable/Declarations/DeclarationTreeBuilder.vb]: Answering my own question:
```VB
result = New String(results.Count -1) {}

Comment (pharring) [inline src/Compilers/VisualBasic/Portable/Declarations/DeclarationTreeBuilder.vb]: I prefer ReDim. It does what it says and it doesn't need the empty curly braces.

**Pull request merged**